### DISCUSSION TITLE
Several configuration file

### DISCUSSION THREAD

**Discussion by quentinalbertone** [vektra/mockery#676] [Q&A]
Hi everyone,

I'm working in a mono-repo environnement which contains several services, I would like to configure mockery through several `.mockery.yaml`. This will help me to prevent a huge file which will mix dozens of interface (and all the issues like merge conflict , etc).

```Bash
. root
|-> ServiceA
|   |- .mockery.yaml
|   |- ohter files and package (without .mockery.yaml)
|-> ServiceB
|   |- .mockery.yaml
|   |- ohter files and package (wihtout .mockery.yaml)
| -> ...
``` 
The goal is tomockery only once at the root of the repo but if it's possible with one call to mockery for each services, I will be happy too.

Thanks

**Comment by LandonTClipp** [comment]
Hi, this has been proposed by someone else, I think there's a decent need for it. The configuration parsing for `packages` is pretty nasty. It's not my proudest code, but it works.

I think all that should be needed here is to add code to the `Config` object that "imports" a parent file. There are a lot of considerations we need to flesh out:

1. What will the syntax look like?
2. How do we handle cyclical imports?
3. Do we allow cyclical imports or should it be a hard error?
4. How should the current implementation of `Config` be added or altered to allow importing?

I think you'll find this task won't be as easy as you might think. I'm open to discussing it further though.

**Reply by quentinalbertone** [reply]
Hi @LandonTClipp , thanks for your answer.
Yes indeed, its not an easy task. Do you already create a issue that I can follow (maybe participate) ?

**Reply by abarreaux-bigblue** [reply]
Hi @LandonTClipp, first of all thanks for the huge work !
I ended up here having the same need as @quentinalbertone, as I'm migrating from a v2 setup where each sub package of my repo has its own setup and I'd like to be able to have a core config at the root and override from sub packages.
Has it been picked up in any way or is this still an open question ?
